File: 448. Find All Numbers Disappeared in an Array/solution.cpp
class Solution {
public:
  vector<int> findDisappearedNumbers(vector<int>& nums) {
    for (int index = 0; index < nums.size(); index++){
      int newIndex = abs(nums[index]) - 1;
      nums[newIndex] = -abs(nums[newIndex]);
    }

    vector<int> result;
    for (int index = 0; index < nums.size(); index++){
      int value = nums[index];
      if (value > 0){
        result.push_back(index + 1);
      }
    }

    return result;
  }
};

File: 448. Find All Numbers Disappeared in an Array/solution.py
class Solution(object):
  def findDisappearedNumbers(self, nums):
    """
    :type nums: List[int]
    :rtype: List[int]
    """
    for index in range(len(nums)):
      newIndex = abs(nums[index]) - 1
      nums[newIndex] = -abs(nums[newIndex])
      
    result = []
    for index, value in enumerate(nums):
      if value > 0:
        result.append(index + 1)
        
    return result